src/main/java/mate/academy/bookstore/service/impl/BookServiceImpl.java

```diff
@@ -17,30 +23,42 @@
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
     private final BookMapper bookMapper;
+    private final CategoryRepository categoryRepository;
 
     @Override
-    public BookDto save(BookRequestDto requestDto) {
+    public BookResponseDto save(BookRequestDto requestDto) {
+        Set<Long> categoryIds = requestDto.getCategoryIds();
+        List<Category> categories = categoryRepository.findAllById(categoryIds);
+
+        if (categoryIds.size() != categories.size()) {
+            Set<Long> availableCategories = categories.stream()
+                    .map(Category::getId)
+                    .collect(Collectors.toSet());
+            categoryIds.removeAll(availableCategories);

```

Comment:
What is the reason to make some removal logic, all you need is compare size to realize that all categories are present